src/rdkafka.h

```diff
@@ -5993,6 +5993,26 @@ typedef rd_kafka_resp_err_t(rd_kafka_interceptor_f_on_thread_exit_t)(
     void *ic_opaque);
 
 
+/**
+ * @brief on_broker_state_chaned() is called just after a broker
+ *        has been created or its state has been changed.
+ *
+ * @param rk The client instance.
+ * @param secproto The security protocol.
+ * @param name The original name of the broker.
+ * @param port The port of the broker.
+ * @param ic_opaque The interceptor's opaque pointer specified in ..add..().
+ *
+ * @returns an error code on failure, the error is logged but otherwise ignored.
+ */
+typedef rd_kafka_resp_err_t(rd_kafka_interceptor_f_on_broker_state_changed_t)(
+    rd_kafka_t *rk,
+    const char *secproto,
+    const char *name,
```

Comment:
I think you want the brokerid too

Reply:
Thank you @edenhill for the quick response.
With the "broker state interceptor", the client application would be able to know whether anything abnormal happened from the brokers, thus could raise an alarm, etc. Another example is, some applications are able to setup multiple connections via TCP, MQ, Kafka, etc, -- there's command for them to query the connections' status, but we miss the info for Kafka.